custom_components/goodwe/number.py

```diff
@@ -47,6 +47,30 @@ async def _set_eco_mode_power(entity: InverterNumberEntity, value: int) -> None:
 
 
 NUMBERS = (
+    GoodweNumberEntityDescription(
+        key="battery_charge_limit",
+        name="Battery charge limit",
+        icon="mdi:battery-charging",
+        entity_category=EntityCategory.CONFIG,
+        unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
```

Comment:
They changed the APIs meanwhile, so native_unit_of_measurement should be used here instead.